gax-java/gax/src/test/java/com/google/api/gax/core/GaxPropertiesTest.java

```diff
@@ -57,10 +58,94 @@ public void testGaxVersion() {
     }
   }
 
+  private static String originalJavaVersion = System.getProperty("java.version");
+  private static String originalJavaVendor = System.getProperty("java.vendor");
+  private static String originalJavaVendorVersion = System.getProperty("java.vendor.version");
+
+  @After
+  public void cleanup() {
+    if (Strings.isNullOrEmpty(originalJavaVersion)) {
+      System.clearProperty("java.version");
+    } else {
+      System.setProperty("java.version", originalJavaVersion);
+    }
+
+    if (Strings.isNullOrEmpty(originalJavaVendor)) {
+      System.clearProperty("java.vendor");
+    } else {
+      System.setProperty("java.vendor", originalJavaVendor);
+    }
+
+    if (Strings.isNullOrEmpty(originalJavaVendorVersion)) {
+      System.clearProperty("java.vendor.version");
+    } else {
+      System.setProperty("java.vendor.version", originalJavaVendorVersion);
+    }
+  }
+
+  @Test
+  public void testGetJavaRuntimeInfo_graalVM() {
```

Comment:
The three test cases below are testing the same scenario from what I can see, we should only keep one.

Reply:
All the three test cases are actual system property values for the various vendors, so just wanted to make sure we are capturing them fine.

Reply:
It's OK to use real life examples, but we also don't want to duplicate the testing effort. Is the purpose of this unit test to test all real life vendors? Or test the functionality of the method? Someone else should be able to understand/maintain/test the functionality of the method without knowing the real life examples. 
From unit tests' perspective, what are the different scenarios between them? From what I can see, they are all testing the same scenario that vendor and vendor version exist, and replacing spaces with underscores.
In addition, we should really mention the test scenario instead of a real life example in the test name.

Reply:
These are just a few "major" vendors, not all real life vendors. All the tests are not testing the same scenario (vendor and vendor version exist, and replacing spaces with underscores). Test for graalvm takes care of cases where only spaces exist, whereas test for temurin takes care of spaces along with unsupported characters, also because we officially support it. I have removed the coretto one. I have also updated the test name for each.

Reply:
Thanks! LGTM